leetcodePython/PrefixSum/prefixsum_523.py

```diff
@@ -0,0 +1,19 @@
+from typing import List
+
+
+class Solution:
+    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
+        # remain 0 have no valid have 0 remain.
+        remain_index = {0: -1}
+        run_sum = 0
+        for i, num in enumerate(nums):
+            run_sum += num
+            remain = run_sum % k
+            if remain not in remain_index:
+                # Set the earliest index that have the remain, to build the maximum sub array.
+                # So size can possibly be >2
+                remain_index[remain] = i
+            else:
+                if i - remain_index[remain] >= 2:
+                    return True
```

Comment:
We want to locate two arrays (all starts from beginning) with the same remainder. They subtract:
1. `Array2_Sum` = K*X + remainder, `Array1_Sum` = K*Y+remainer
2. `Array2_Sum` - `Array1_Sum` = K*Z
3. `Array2` - `Array1` forms a valid sub array